google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java

```diff
@@ -80,26 +84,51 @@ private JsonStreamWriter(Builder builder)
 
   /**
    * Writes a JSONArray that contains JSONObjects to the BigQuery table by first converting the JSON
-   * data to protobuf messages, then using StreamWriter's append() to write the data.
+   * data to protobuf messages, then using StreamWriter's append() to write the data. If there is a
+   * schema update, the current StreamWriter is closed. A new StreamWriter is created with the
+   * updated TableSchema.
    *
    * @param jsonArr The JSON array that contains JSONObjects to be written
    * @return ApiFuture<AppendRowsResponse> returns an AppendRowsResponse message wrapped in an
    *     ApiFuture
    */
-  public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr) {
+  public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr)
+      throws IOException, DescriptorValidationException {
     return append(jsonArr, -1);
   }
 
   /**
    * Writes a JSONArray that contains JSONObjects to the BigQuery table by first converting the JSON
-   * data to protobuf messages, then using StreamWriter's append() to write the data.
+   * data to protobuf messages, then using StreamWriter's append() to write the data. If there is a
+   * schema update, the current StreamWriter is closed. A new StreamWriter is created with the
+   * updated TableSchema.
    *
    * @param jsonArr The JSON array that contains JSONObjects to be written
    * @param offset Offset for deduplication
    * @return ApiFuture<AppendRowsResponse> returns an AppendRowsResponse message wrapped in an
    *     ApiFuture
    */
-  public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset) {
+  public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset)
+      throws IOException, DescriptorValidationException {
+    // Handle schema updates
+    if (this.streamWriter.getUpdatedSchema() != null) {
+      TableSchema updatedSchema = this.streamWriter.getUpdatedSchema();
+      // Close the StreamWriter
+      this.streamWriter.close();
+      // Update JsonStreamWriter's TableSchema and Descriptor
+      this.tableSchema = updatedSchema;
+      this.descriptor =
+          BQTableSchemaToProtoDescriptor.convertBQTableSchemaToProtoDescriptor(updatedSchema);
+      // Create a new underlying StreamWriter with the updated TableSchema and Descriptor
+      this.streamWriter =
+          streamWriterBuilder
+              .setUpdatedTableSchema(this.tableSchema)
+              .setWriterSchema(ProtoSchemaConverter.convert(this.descriptor))
+              .build();
+      // Clear the updateSchema field on the new underlying StreamWriter
+      this.streamWriter.clearUpdateSchema();
```

Comment:
Since this streamWriter is newly created, we don't need to do this.

Reply:
Okay. removing the redundant clear method.